tests/test_matrix_utils_connect_rooms.py

```
@@ -0,0 +1,296 @@
+            "homeserver": "https://matrix.org",
+            "access_token": "test_token",
+            "user_id": "@test:matrix.org",
+            "device_id": "test_device_id",
+        }
+
+        # Mock the AsyncClient instance
+        mock_client_instance = MagicMock()
+        mock_client_instance.rooms = {}
+
+        # Create proper async mock methods
+        async def mock_whoami():
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `mock_whoami`

(ANN202)